File: 3740. minimum-distance-between-three-equal-elements-i.py
from typing import List

class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        pos = {}
        
        for i, num in enumerate(nums):
            if num not in pos:
                pos[num] = []
            pos[num].append(i)
        
        ans = float('inf')
        
        for indices in pos.values():
            if len(indices) < 3:
                continue
            
            for i in range(len(indices) - 2):
                distance = 2 * (indices[i + 2] - indices[i])
                ans = min(ans, distance)
        
        return -1 if ans == float('inf') else ans

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [1,2,1,1,3]
    output1 = sol.minimumDistance(nums1)
    print(f"Example 1: nums={nums1}")
    print(f"Output: {output1}")
    print(f"Expected: 6")
    print()
    
    # Example 2
    nums2 = [1,1,2,3,2,1,2]
    output2 = sol.minimumDistance(nums2)
    print(f"Example 2: nums={nums2}")
    print(f"Output: {output2}")
    print(f"Expected: 8")
    print()
    
    # Example 3
    nums3 = [1]
    output3 = sol.minimumDistance(nums3)
    print(f"Example 3: nums={nums3}")
    print(f"Output: {output3}")
    print(f"Expected: -1")

File: 3740. minimum-distance-between-three-equal-elements-i.java
import java.util.*;

class Solution {
    public int minimumDistance(int[] nums) {
        Map<Integer, List<Integer>> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            map.computeIfAbsent(nums[i], k -> new ArrayList<>()).add(i);
        }
        
        int ans = Integer.MAX_VALUE;
        
        for (List<Integer> indices : map.values()) {
            if (indices.size() < 3) continue;
            
            for (int i = 0; i + 2 < indices.size(); i++) {
                int distance = 2 * (indices.get(i + 2) - indices.get(i));
                ans = Math.min(ans, distance);
            }
        }
        
        return ans == Integer.MAX_VALUE ? -1 : ans;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {1,2,1,1,3};
        int output1 = sol.minimumDistance(nums1);
        System.out.println("Example 1: nums=[1,2,1,1,3]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 6");
        System.out.println();
        
        // Example 2
        int[] nums2 = {1,1,2,3,2,1,2};
        int output2 = sol.minimumDistance(nums2);
        System.out.println("Example 2: nums=[1,1,2,3,2,1,2]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 8");
        System.out.println();
        
        // Example 3
        int[] nums3 = {1};
        int output3 = sol.minimumDistance(nums3);
        System.out.println("Example 3: nums=[1]");
        System.out.println("Output: " + output3);
        System.out.println("Expected: -1");
    }
}